test/core/verbencoding.js

```diff
@@ -0,0 +1,134 @@
+describe("Core htmx AJAX Verbs with json encoding", function() {
+    beforeEach(function () {
+        this.server = makeServer();
+        clearWorkArea();
+
+    });
+    afterEach(function () {
+        this.server.restore();
+        clearWorkArea();
+        // if we don't set this, then all the following test MAY 
+        // inherit incorrect encoding, and fail. 
+        htmx.config.defaultEncoding="xml";
+    });
+
+    it('handles basic post properly', function () {
+        var jsonResponseBody = JSON.stringify({});
+        this.server.respondWith("POST", "/test", jsonResponseBody);
+        var div = make('<div hx-post="/test encoding:json">click me</div>');
+        div.click();
+        this.server.respond();
+        this.server.lastRequest.response.should.equal("{}");
+    })
+
+    it('handles basic put properly', function () {
+        var jsonResponseBody = JSON.stringify({});
+        this.server.respondWith("PUT", "/test", jsonResponseBody);
+        var div = make('<div hx-put="/test encoding:json">click me</div>');
+        div.click();
+        this.server.respond();
+        this.server.lastRequest.response.should.equal("{}");
+    })
+
+    it('handles basic patch properly', function () {
+        var jsonResponseBody = JSON.stringify({});
+        this.server.respondWith("PATCH", "/test", jsonResponseBody);
+        var div = make('<div hx-patch="/test encoding:json">click me</div>');
+        div.click();
+        this.server.respond();
+        this.server.lastRequest.response.should.equal("{}");
+    })
+
+    it('handles basic delete properly', function () {
+        var jsonResponseBody = JSON.stringify({});
+        this.server.respondWith("DELETE", "/test", jsonResponseBody);
+        var div = make('<div hx-delete="/test encoding:json">click me</div>');
+        div.click();
+        this.server.respond();
+        this.server.lastRequest.response.should.equal("{}");
+    })
+
+    it('handles post with form parameters', function () {
+
+        this.server.respondWith("POST", "/test", function (xhr) {
+            var values = JSON.parse(xhr.requestBody);
```

Comment:
I think it is worth testing the parameter values here as well

Reply:
ok, will add it.

Reply:
Added: 
```
            values["username"].should.be.equal("joe");
            values["password"].should.be.equal("123456");
```
to 5 tests.